src/test_bargain.c

```diff
@@ -0,0 +1,364 @@
+    {
+        return 1;
+    }
+    return 0;
+}
+
+int compare_basket_items(basket_item_s x, basket_item_s y)
+{
+    if (strcmp(x.name, y.name) == 0 && x.size == y.size && x.unit == y.unit)
+    {
+        return 1;
+    }
+    return 0;
+}
+
+int compare_store_items(store_item_s x, store_item_s y)
+{
+    if (strcmp(x.name, y.name) == 0 && x.price == y.price && x.price_per_unit == y.price_per_unit && x.size == y.size && x.unit == y.unit)
+    {
+        return 1;
+    }
+    return 0;
+}
+
+int compare_stores_structs(store_s x, store_s y)
+{
+
+    if (x.items_count != y.items_count)
+    {
+        return 0;
+    }
+
+    if (x.found_items_count != y.found_items_count)
+    {
+        return 0;
+    }
+
+    if (x.missing_items_count != y.missing_items_count)
+    {
+        return 0;
+    }
+
+    for (int i = 0; i < x.items_count; i++)
+    {
+
+        if (!compare_store_items(x.items[i], y.items[i]))
+        {
+            return 0;
+        }
+    }
+
+    for (int i = 0; i < x.found_items_count; i++)
+    {
+
+        if (!compare_found_items(x.found_items[i], y.found_items[i]))
+        {
+            return 0;
+        }
+    }
+
+    for (int i = 0; i < x.missing_items_count; i++)
+    {
+        if (!compare_basket_items(x.missing_items[i], y.missing_items[i]))
+        {
+            return 0;
+        }
+    }
+    return 1;
+}
+
+// int test_stores(store_s *stores, int store_count)
+// {
+//     int expected_stores_count = 3;
+//     store_s *expected_stores = get_expected_stores();
+//     int store_counts_match = (expected_stores_count == store_count);
+//     printf("Store count matches: %d == %d?\n", expected_stores_count, store_count);
+//     assert(store_counts_match);
+//     for (int i = 0; i < 3; i++)
+//     {
+//         int test_store_result_match_expectation = (!compare_stores_structs(stores[i], expected_stores[i]));
+//         assert(test_store_result_match_expectation);
+//     }
+//     return 1;
+// }
+
+int test_bargain_make_assertions(store_s *stores, int store_count)
+{
+    // test_stores(stores, store_count);
+    return 0;
+}
+
+int test_bargain_get_basket(basket_item_s *basket)
+{
+    // basket_item_s item1 = {
+    //     .name = "Milk",
+    //     .size = 1,
+    //     .unit = LITERS,
+    // };
+
+    // basket_item_s item2 = {
+    //     .name = "Beef",
+    //     .size = 2,
+    //     .unit = KILOGRAMS,
+    // };
+
+    // basket_item_s item3 = {
+    //     .name = "Noodles",
+    //     .size = 3,
+    //     .unit = UNITS,
+    // };
+
+    // basket_item_s item4 = {
+    //     .name = "Snickers",
+    //     .size = 4,
+    //     .unit = UNITS,
+    // };
+
+    // basket_item_s item5 = {
+    //     .name = "Pudding",
+    //     .size = 5,
+    //     .unit = UNITS,
+    // };
+
+    basket[0] = item1;
+    basket[1] = item2;
+    basket[2] = item3;
+    basket[3] = item4;
+    basket[4] = item5;
+
+    return 5;
+}
```

Comment:
From a quick glance, this still looks fucked up, so I'm choosing to ignore this for the sake of my own sanity

Reply:
Yep, it's gimped